spec/javascripts/clear_form_radio_buttons_spec.js

```diff
@@ -0,0 +1,52 @@
+//= require jquery
+//= require jasmine-jquery
+
+var input = $('<div class="js-clearRadioButtons">' +
+  '<label class="block-label date-box selected" for="visit_slot_granted">' +
+  '<input checked="checked" type="radio" value="2017-01-24T14:00/16:00" name="visit[slot_granted]" id="visit_slot_granted">' +
+  '</label>' +
+  '<a href="#" id="clear" class="js-clearRadioButton" data-target="visit[slot_granted]">Clear selection</a>' +
+  '</div>');
+
+describe('Clear radio button', function() {
+
+  beforeEach(function() {
+    $('body').append(input);
+  });
+
+  describe('Init', function() {
+    beforeEach(function() {
+      spyOn(moj.Modules.clearRadioButtons, 'bindEvents');
+    });
+    it('should call `this.bindEvents`', function() {
+      moj.Modules.clearRadioButtons.init();
+      expect(moj.Modules.clearRadioButtons.bindEvents).toHaveBeenCalled();
+    });
+  });
+
+  describe('should be checked', function() {
+    it('input should be checked', function() {
+      expect($('#visit_slot_granted').is(':checked')).toBe(true);
+    });
+
+    it('label should have `selected` class', function() {
+      expect($('#visit_slot_granted').parent('label')).toHaveClass('selected');
+    });
+  });
+
+  describe('should be unchecked', function() {
+    beforeEach(function() {
+      moj.Modules.clearRadioButtons.init();
+      spyOnEvent('.js-clearRadioButton', 'click');
```

Comment:
I see, i don't really understand the value of that test. To me it seems you're that triggering an event triggers the event. You would want to spy on an event if it is triggered by something else like from a calling function.